services/pull/pull.go

```diff
@@ -57,6 +57,41 @@ func NewPullRequest(repo *models.Repository, pull *models.Issue, labelIDs []int6
 
 	notification.NotifyNewPullRequest(pr)
 
+	// add first push codes command
+	baseGitRepo, err := git.OpenRepository(pr.BaseRepo.RepoPath())
+	if err != nil {
+		return err
+	}
+	defer baseGitRepo.Close()
+
+	compareInfo, err := baseGitRepo.GetCompareInfo(pr.BaseRepo.RepoPath(),
+		pr.BaseBranch, pr.GetGitRefName())
+	if err != nil {
+		return err
+	}
+
+	if compareInfo.Commits.Len() > 0 {
```

Comment:
Ahem, this should always be true; otherwise it's a bug. 😅

Reply:
Maybe it's necessary, can see https://drone.gitea.io/go-gitea/gitea/24427/2/10.
But I don't know the reason ...